config/yamlflag_test.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"testing"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type yamlFlagTestConfig struct {
+	Foo string
+	Bar []string
+	Baz int
+	Qux *yamlFlagTestConfig
+}
+
+func TestYamlFlag(t *testing.T) {
+	t.Run("set", func(t *testing.T) {
+		cfg := struct {
+			Field *yamlFlagTestConfig
+		}{}
+
+		f := newYamlFlag(&cfg.Field)
+		v := `{foo: hello, bar: [world, "1"], baz: 2, qux: {baz: 3}}`
+		f.Set(v)
+
+		assert.Equal(t, "hello", cfg.Field.Foo)
+		assert.Equal(t, []string{"world", "1"}, cfg.Field.Bar)
+		assert.Equal(t, 2, cfg.Field.Baz)
+		assert.Equal(t, 3, cfg.Field.Qux.Baz)
+
+		assert.Equal(t, v, f.String())
+	})
+
+	t.Run("set empty", func(t *testing.T) {
+		cfg := struct {
+			Field *yamlFlagTestConfig
+		}{}
+
+		f := newYamlFlag(&cfg.Field)
+		f.Set("")
+
+		assert.Equal(t, &yamlFlagTestConfig{}, cfg.Field)
+		assert.Equal(t, "", f.String())
+	})
+
+	t.Run("set empty object", func(t *testing.T) {
+		cfg := struct {
+			Field *yamlFlagTestConfig
+		}{}
+
+		f := newYamlFlag(&cfg.Field)
+		f.Set("{}")
+
+		assert.Equal(t, &yamlFlagTestConfig{}, cfg.Field)
+		assert.Equal(t, "{}", f.String())
+	})
+
+	t.Run("unmarshal YAML", func(t *testing.T) {
```

Comment:
I think we can another test case for non-valid yaml, wdyt?

Reply:
Makes sense, added.